tests/test_nats_sts.py

```diff
@@ -0,0 +1,119 @@
+        """Test that nats statefulset renders good metrics exporter."""
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only=["charts/nats/templates/statefulset.yaml"],
+            values={
+                "nats": {
+                    "exporter": {
+                        "enabled": True,
+                        "resources": {"requests": {"cpu": "234m"}},
+                    },
+                    "nats": {"resources": {"requests": {"cpu": "123m"}}},
+                },
+            },
+        )
+
+        assert len(docs) == 1
+        containers = docs[0]["spec"]["template"]["spec"]["containers"]
+        assert len(containers) == 2
+        c_by_name = {c["name"]: c for c in containers}
+        assert c_by_name["nats"]["resources"]["requests"]["cpu"] == "123m"
+        assert c_by_name["metrics"]["resources"]["requests"]["cpu"] == "234m"
+
+    def test_nats_statefulset_with_affinity_and_tolerations(self, kube_version):
+        """Test that nats statefulset renders proper nodeSelector, affinity, and tolerations"""
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only=["charts/nats/templates/statefulset.yaml"],
+            values={
+                "nats": {
+                    "nodeSelector": {"role": "astro"},
+                    "affinity": {
+                        "nodeAffinity": {
+                            "requiredDuringSchedulingIgnoredDuringExecution": {
+                                "nodeSelectorTerms": [
+                                    {
+                                        "matchExpressions": [
+                                            {
+                                                "key": "astronomer.io/multi-tenant",
+                                                "operator": "In",
+                                                "values": ["false"],
+                                            }
+                                        ]
+                                    }
+                                ]
+                            }
+                        }
+                    },
+                    "tolerations": [
+                        {
+                            "effect": "NoSchedule",
+                            "key": "astronomer",
+                            "operator": "Exists",
+                        }
+                    ],
+                },
+            },
+        )
+
+        assert len(docs) == 1
+        spec = docs[0]["spec"]["template"]["spec"]
+        assert spec["nodeSelector"] != {}
+        assert spec["nodeSelector"]["role"] == "astro"
+        assert spec["affinity"] != {}
+        assert (
+            len(
+                spec["affinity"]["nodeAffinity"][
+                    "requiredDuringSchedulingIgnoredDuringExecution"
+                ]["nodeSelectorTerms"]
+            )
+            == 1
+        )
+        assert len(spec["tolerations"]) > 0
```

Comment:
Maybe here you could directly assert that spec["tolerations"] is the same as the one you've passed to the template.

```python
values = {} # the values defined above
docs = render_chart(values=values) #...

assert spec["tolerations"] = values["nats"]["tolerations"]
```

It would make sure that even if in the futur we add a default tolerations or injected from another variable, we're still testing the right behavior (injecting variables from the chart level values).